server/src/main/java/dev/slimevr/config/CurrentVRConfigConverter.java

```diff
@@ -18,188 +19,212 @@ public ObjectNode convert(
+				if (skeletonNode != null) {
+					ObjectNode offsetsNode = (ObjectNode) skeletonNode.get("offsets");
+					if (offsetsNode != null) {
+						if (offsetsNode.get("chestOffset") != null) {
+							offsetsNode
+								.set(
+									"chestOffset",
+									new FloatNode(-offsetsNode.get("chestOffset").floatValue())
+								);
+						}
+						if (offsetsNode.get("hipOffset") != null) {
+							offsetsNode
+								.set(
+									"hipOffset",
+									new FloatNode(-offsetsNode.get("hipOffset").floatValue())
+								);
+						}
+						if (offsetsNode.get("elbowOffset") != null) {
+							offsetsNode
+								.set(
+									"elbowOffset",
+									new FloatNode(-offsetsNode.get("elbowOffset").floatValue())
+								);
+						}
+					}
+				}
+			}
+			if (version < 8) {
+				// reset > fullReset, quickReset > yawReset
+				ObjectNode keybindingsNode = (ObjectNode) modelData.get("keybindings");
+				if (keybindingsNode != null) {
+					keybindingsNode.set("fullResetBinding", keybindingsNode.get("resetBinding"));
+					keybindingsNode
+						.set("yawResetBinding", keybindingsNode.get("quickResetBinding"));
+					keybindingsNode
+						.set("mountingResetBinding", keybindingsNode.get("resetMountingBinding"));
+					if (keybindingsNode.get("resetDelay") != null) {
+						keybindingsNode.set("fullResetDelay", keybindingsNode.get("resetDelay"));
+						keybindingsNode
+							.set("yawResetDelay", keybindingsNode.get("quickResetDelay"));
+						keybindingsNode
+							.set("mountingResetDelay", keybindingsNode.get("resetMountingDelay"));
+					}
+				}
 
-			ObjectNode tapDetectionNode = (ObjectNode) modelData.get("tapDetection");
-			if (tapDetectionNode != null) {
-				tapDetectionNode.set("yawResetDelay", tapDetectionNode.get("quickResetDelay"));
-				tapDetectionNode.set("fullResetDelay", tapDetectionNode.get("resetDelay"));
-				tapDetectionNode.set("yawResetEnabled", tapDetectionNode.get("quickResetEnabled"));
-				tapDetectionNode.set("fullResetEnabled", tapDetectionNode.get("resetEnabled"));
-				tapDetectionNode.set("yawResetTaps", tapDetectionNode.get("quickResetTaps"));
-				tapDetectionNode.set("fullResetTaps", tapDetectionNode.get("resetTaps"));
+				ObjectNode tapDetectionNode = (ObjectNode) modelData.get("tapDetection");
+				if (tapDetectionNode != null) {
+					tapDetectionNode.set("yawResetDelay", tapDetectionNode.get("quickResetDelay"));
+					tapDetectionNode.set("fullResetDelay", tapDetectionNode.get("resetDelay"));
+					tapDetectionNode
+						.set("yawResetEnabled", tapDetectionNode.get("quickResetEnabled"));
+					tapDetectionNode.set("fullResetEnabled", tapDetectionNode.get("resetEnabled"));
+					tapDetectionNode.set("yawResetTaps", tapDetectionNode.get("quickResetTaps"));
+					tapDetectionNode.set("fullResetTaps", tapDetectionNode.get("resetTaps"));
+				}
 			}
+		} catch (Exception e) {
+			LogManager.severe("Error during config migration: " + e);
 		}
```

Comment:
If anything goes wrong, do we make a backup of the original config? We most definitely should. If it's not viable to do it on an error, then do it before any config migration is done.

Reply:
Not sure that’s possible with how it is currently set, but Uriel wants to rewrite the whole config system or something anyways…

Reply:
I think butter is saying to just copy the config file and put it into a .old one

Reply:
I know, but does the class have access to the file itself? All I see is that it has access to the content… not even the file path, and we can’t access vrServer because it’s in lateinit

Reply:
sigh, yeah lets leave it for the rewrite

Reply:
Unfortunate, probably "okay" then as long as it's fairly reliable